### PR TITLE
#833 Fix uses of Image when underlying array is not C-contiguous

### PR DESCRIPTION
**Pull request by rmjarvis**
In #799, I updated the Image class to work with arbitrary steps between the values in both directions, i.e. not require the underlying array to be C-contiguous.  However, I missed some uses that depended on the C-contiguousness of the array.  Since that's usually what we have, it didn't trigger any errors in the tests.

Anyway, Josh ran across one, which he reported in #833.  I fixed that and added unit tests that would have caught the problem.  There were also problems with the addNoise implementations, so I fixed that and added tests there as well.

SBInterpolatedImage serialize also used the error-prone iterators, although it always uses a C-contiguous image, so it wasn't ever a problem.  But I switched it anyway and removed the Image iterator methods.

### CODE REVIEW

File: tests/test_hsm.py [hunk 1]
```diff
@@ -791,6 +791,37 @@ def test_ksb_sig():
...
+
+    This test was inspired by a bug report in issue #833.
+    """
+    gal = galsim.Gaussian(sigma=1.3).shear(g1=0.1, g2=0.2)
+
+    # First a normal C-contiguous image as build by drawImage.
+    img = gal.drawImage(nx=64, ny=64, scale=0.2)
+    meas_shape1 = galsim.hsm.FindAdaptiveMom(img).observed_shape
+    print(meas_shape1)
+    np.testing.assert_almost_equal(meas_shape1.g1, 0.1, decimal=3)
```
Comment (rmandelb) [inline tests/test_hsm.py]: Put a useful ```err_msg``` on all of these?

**Comment by rmjarvis** [comment]
> this isn't in CHANGELOG, but I assume that was by design since the updates are just fixing bugs that happened after the last tagged release.

Yes.  This wasn't broken in 1.4.  So no need to count it as a bug fix as far as the CHANGELOG is concerned.

File: tests/test_hsm.py (hunk 1 above)
Comment (rmandelb) [inline tests/test_hsm.py]: Thanks.

**Comment by jmeyers314** [comment]
@rmjarvis , this one looks good to me too.

**Pull request merged**